### PR TITLE
fixed password endline bug

### LINKED ISSUE #5444: Configsdb: postgres auth fails after v1.15.0
**Describe the bug**
Configs: postgres auth fails after https://github.com/cortexproject/cortex/commit/1fed98a142b5077c3776e8d3def9992a8c8ae724

Configs is not ignoring the end line in password file anymore.

**To Reproduce**
1. Start Postgres with password enabled

```
docker run --rm -p 5432:5432 -e POSTGRES_PASSWORD=pass postgres:12-alpine -d postgres
```
2.  Create password file with end line
```
echo pass > password
```
3. Start cortex with target configs
```
docker run --network=host -v $PWD/password:/password cortexproject/cortex:master-1fed98a -configs.database.uri=postgres://postgres@127.0.0.1/postgres?sslmode=disable -configs.database.password-file=password -target=configs
```
4. See error authenticating:

```
level=info ts=2023-07-07T16:31:56.030503Z caller=main.go:194 msg="Starting Cortex" version="(version=1.14.0, branch=master, revision=1fed98a)"
level=info ts=2023-07-07T16:31:56.03981Z caller=server.go:306 http=[::]:80 grpc=[::]:9095 msg="server listening on addresses"
level=warn ts=2023-07-07T16:31:56.056675Z caller=postgres.go:62 msg="db connection not established, retrying..." err="pq: password authentication failed for user \"postgres\""
```

Previous commit works fine 
https://github.com/cortexproject/cortex/commit/a602a7f2825fcdd1336d16a170d551bee7a49c8b

```
docker run --network=host -v $PWD/password:/password cortexproject/cortex:master-a602a7f -configs.database.uri=postgres://postgres@127.0.0.1/postgres?sslmode=disable -configs.database.password-file=password -target=configs
```

**Expected behavior**
Auth should work. We should remove end lines after the password, so password works. I doubt anybody is using end lines as part of the password.

**Additional Context**

lib/pq is more pedantic after https://github.com/lib/pq/commit/072e83d238c297c71cab11154695af301010031e
and probably for good reasons.

The workaround is to create the password file without the end line:
```
echo -n pass > password
...
```

### PR DESCRIPTION
**Pull request by moki1202**
**Which issue(s) this PR fixes**:
Fixes #5444 

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by friedrichg — CHANGES_REQUESTED** [review]
And also mention the fix in the changelog

File: pkg/configs/db/db.go [hunk 1]
```diff
@@ -77,7 +78,8 @@ func New(cfg Config) (DB, error) {
 		if err != nil {
 			return nil, fmt.Errorf("Could not read database password file: %v", err)
 		}
-		u.User = url.UserPassword(u.User.Username(), string(passwordBytes))
+		password := strings.TrimSpace(string(passwordBytes))
+		u.User = url.UserPassword(u.User.Username(), password)
```
Comment (friedrichg) [inline pkg/configs/db/db.go:82]: Looks good, but it's not complete. Ideally for every change we modify or add a tests.

you can extract a function that returns url `u` and error as needed and test that function.

Comment (moki1202) [reply pkg/configs/db/db.go:82]: @friedrichg apologies for the late response, I can't seem to find a test file for this file. Should I create a new one or am I missing something?

**Comment by moki1202** [comment]
@friedrichg Please let me know if this is what you asked for. If correct, I'll add a few more test cases.

File: pkg/configs/db/dbtest/unit.go [hunk 2]
```diff
@@ -4,6 +4,9 @@
 package dbtest
```
Comment (friedrichg) [inline pkg/configs/db/dbtest/unit.go:4]: Don't modify this file, this is part of integration tests

File: pkg/configs/db/dbtest/unit.go [hunk 3]
```diff
@@ -26,3 +29,48 @@ func Setup(t *testing.T) db.DB {
 func Cleanup(t *testing.T, database db.DB) {
 	require.NoError(t, database.Close())
 }
+
+func TestUserPasswordFromPasswordFile(t *testing.T) {
```
Comment (friedrichg) [inline pkg/configs/db/dbtest/unit.go:33]: create a file called db_test.go next to db.go and move this test there.

File: pkg/configs/db/dbtest/unit.go [hunk 4]
```diff
@@ -26,3 +29,48 @@ func Setup(t *testing.T) db.DB {
...
+
+	tempFile, _ := ioutil.TempFile("", "testpassword")
+	defer tempFile.Close()
+	defer os.Remove(tempFile.Name())
```
Comment (friedrichg) [inline pkg/configs/db/dbtest/unit.go:37]: defer functions have a specific order.
In this case we are removing and then closing. We want the opposite.
https://go.dev/tour/flowcontrol/13#:~:text=Deferred%20function%20calls%20are%20pushed,in%2Dfirst%2Dout%20order.

**Review by friedrichg — CHANGES_REQUESTED** [review]
Just a couple of nits

File: pkg/configs/db/db.go [hunk 5]
```diff
@@ -70,14 +86,11 @@ func New(cfg Config) (DB, error) {
 	}
 
 	if len(cfg.PasswordFile) != 0 {
-		if u.User == nil {
-			return nil, fmt.Errorf("--database.password-file requires username in --database.uri")
-		}
-		passwordBytes, err := os.ReadFile(cfg.PasswordFile)
+		updatedUrl, err := SetUserPassword(u, cfg.PasswordFile)
```
Comment (friedrichg) [inline pkg/configs/db/db.go:89]:
```suggestion
		updatedURL, err := SetUserPassword(u, cfg.PasswordFile)
```
So it passes linting

**Comment by moki1202** [comment]
@friedrichg done! shoould I add few more testcases here?

**Comment by moki1202** [comment]
@friedrichg should I add few more testcases here? please let me know if this PR needs further tweaks.

**Comment by friedrichg** [comment]
@moki1202 the lint has to pass and the changelog entry is missing

**Pull request closed without merging**